ml-agents/mlagents/trainers/trainer_controller.py

```diff
@@ -270,30 +283,35 @@ def take_step(self, env, curr_info: AllBrainInfo):
             take_action_text[brain_name] = action_info.text
             take_action_value[brain_name] = action_info.value
             take_action_outputs[brain_name] = action_info.outputs
+        time_start_step = time()
         new_info = env.step(
             vector_action=take_action_vector,
             memory=take_action_memories,
             text_action=take_action_text,
             value=take_action_value
         )
-
+        delta_time_step = time() - time_start_step
         for brain_name, trainer in self.trainers.items():
+            if brain_name in self.trainer_metrics:
+                self.trainer_metrics[brain_name].add_delta_step(delta_time_step)
```

Comment:
Why not `self.trainers[brain_name].trainer_metrics` ?

Reply:
Can you explain this a bit more? Are you saying why have the if condition?